File: Selection Sort/Selection Sort.py
import sys

def selectionSort(a, n):
    countCmpOps = 0 #비교 연산자 실행 횟수
    countSwap = 0 # swap 함수 실행 횟수

    for i in range(n-1):
        jMin = i

        for j in range(i+1, n):
            if(a[j] < a[jMin]):
                jMin = j
                countCmpOps += 1
            if(jMin != 1):
                t = a[jMin]
                a[jMin] = a[j]
                a[j] = t
                countSwap += 1

    print(countCmpOps + " " + countSwap)



numTestCases = int(input())

for i in range(0, numTestCases):
    num = list(map(int, sys.stdin.readline().split()))
    a = [0 for k in range(num[0])]
    
    for j in range(0, num[0]):
       a[j] = num[j+1]
        
    selectionSort(a, num[0])

File: Selection Sort/Selection Sort.cpp
#include <iostream>

using namespace std;
void selectionSort(int a[], int n) {
    int countCmpOps = 0; // 비교 연산자 실행 횟수
    int countSwap = 0; // swap 함수 실행 횟수

    for (int i = 0; i < n - 1; i++) {
        int jMin = i;

        for (int j = i + 1; j < n; j++) {
            countCmpOps++;
            if (a[j] < a[jMin]) {
                jMin = j;
            }
        }
        if (jMin != i) {
            int t = a[jMin];
            a[jMin] = a[i];
            a[i] = t;
            countSwap++;
        }
    }

    cout << countCmpOps << " " << countSwap << endl;

}

int main() {
    int numTestCases;
    int num;
    const int MAX = 1000;

    cin >> numTestCases;

    for (int i = 0; i < numTestCases; i++) {
        cin >> num;
        int a[MAX];
        for (int j = 0; j < num; j++) {
            cin >> a[j];
        }

        selectionSort(a, num);
    }

    return 0;
}